onnxruntime/core/xnnpack/schema/xnnpack_onnx_defs.cc

```diff
@@ -0,0 +1,281 @@
+    filter_size = (filter_size - 1) * dilation_rate + 1;
+  }
+  if (input_size + 1 <= filter_size) {
+    *output_size = -1;
+    return OnnxStatus(StatusCategory::NONE, StatusCode::FAIL);
+  }
+
+  *output_size = input_size - filter_size + stride;
+  *output_size = *output_size / stride;
+  return ::ONNX_NAMESPACE::Common::Status::OK();
+}
+
+// padding_mode: 0, valid. 1, same
+OnnxStatus ConvShapeInference(const ::ONNX_NAMESPACE::TensorShapeProto_Dimension& batch_shape, ptrdiff_t in_height,
+                              ptrdiff_t in_width, ptrdiff_t in_channels,
+                              const ::ONNX_NAMESPACE::TensorShapeProto_Dimension& out_channels, ptrdiff_t filter_height,
+                              ptrdiff_t filter_width, ptrdiff_t in_channels1, uint32_t strides_h, uint32_t strides_w,
+                              uint32_t dilation_h, uint32_t dilation_w, int padding_mode,
+                              ::ONNX_NAMESPACE::TensorShapeProto_Dimension** output) {
+  if (in_channels != in_channels1) {
+    return OnnxStatus(StatusCategory::NONE, StatusCode::FAIL);
+  }
+
+  *output[0] = batch_shape;
+  ptrdiff_t output1, output2;
+  if (padding_mode == 1) {
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeSame(in_height, strides_h, &output1));
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeSame(in_width, strides_w, &output2));
+  } else if (padding_mode == 0) {
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeValid(in_height, strides_h, filter_height, dilation_h, &output1));
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeValid(in_width, strides_w, filter_width, dilation_w, &output2));
+  } else {
+    return OnnxStatus(StatusCategory::NONE, StatusCode::FAIL, "Invalid padding mode.");
+  }
+
+  *output[3] = out_channels;
+  output[1]->set_dim_value(output1);
+  output[2]->set_dim_value(output2);
+  return ::ONNX_NAMESPACE::Common::Status::OK();
+}
+
+OnnxStatus XnnPackConvShapeInferImpl(const ::ONNX_NAMESPACE::TensorShapeProto& input_shape,
+                                     const ::ONNX_NAMESPACE::TensorShapeProto& weight_shape, uint32_t input_padding_top,
+                                     uint32_t input_padding_right, uint32_t input_padding_bottom,
+                                     uint32_t input_padding_left, uint32_t subsampling_height,
+                                     uint32_t subsampling_width, uint32_t dilation_h, uint32_t dilation_w,
+                                     int padding_mode, ::ONNX_NAMESPACE::TensorShapeProto* final_output_shape) {
+  if (input_shape.dim_size() != 4) {
+    return OnnxStatus(StatusCategory::NONE, StatusCode::FAIL, "Input tensor must have 4 dimensions");
+  }
+
+  if (weight_shape.dim_size() != 4) {
+    return OnnxStatus(StatusCategory::NONE, StatusCode::FAIL, "Weight tensor must have 4 dimensions");
+  }
+  for (int i = 1; i != 3; ++i) {
+    if (!input_shape.dim(i).has_dim_value()) {
+      return OnnxStatus(StatusCategory::NONE, StatusCode::FAIL, "Only the first dim(batch size) can be unknown");
```

Comment:
>Only the first dim(batch size) can be unknown [](http://example.com/codeflow?start=65&length=45)

H and W can be unknown even filter is constant, but it is rare though

Reply:
yolov3 is such a model, and I can't make xnnpack support it.